front/lib/swr/assistants.ts

```diff
@@ -226,7 +226,7 @@ export function useSlackChannelsLinkedWithAgent({
     `/api/w/${workspaceId}/data_sources/${dataSourceName}/managed/slack/channels_linked_with_agent`,
```

Comment:
You are missing:
```suggestion
    dataSourceName ? `/api/w/${workspaceId}/data_sources/${dataSourceName}/managed/slack/channels_linked_with_agent` : null,
```

Reply:
I don't think it's required is it?

Reply:
Not necessary with the new `disabled` implementation, but this is generally the pattern we use to avoid "broken" cache keys.